File: base/android/java/src/org/chromium/base/EventLog.java
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.base;

/**
 * A simple interface to Android's EventLog to be used by native code.
 */
@JNINamespace("base::android")
public class EventLog {

    @CalledByNative
    public static void writeEvent(int tag, int value) {
        android.util.EventLog.writeEvent(tag, value);
    }
}

File: base/android/event_log.cc
// Copyright 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/android/event_log.h"
#include "jni/EventLog_jni.h"

namespace base {
namespace android {

void EventLogWriteInt(int tag, int value) {
  Java_EventLog_writeEvent(AttachCurrentThread(), tag, value);
}

bool RegisterEventLog(JNIEnv* env) {
  return RegisterNativesImpl(env);
}

}  // namespace android
}  // namespace base
